Remotion/Data/DomainObjects.UnitTests/Mapping/ClassDefinitionTest.cs

```diff
@@ -545,48 +546,53 @@ public void GetAllPropertyDefinitions_ReadOnly ()
     [Test]
     public void AddPropertyToOtherClass ()
     {
-      var companyClass = ClassDefinitionObjectMother.CreateClassDefinition("Company");
-      var orderClass = ClassDefinitionObjectMother.CreateClassDefinition("Order");
+      var companyClass = ClassDefinitionObjectMother.CreateClassDefinition("Company", typeof(Company));
```

Comment:
I don't think that it is worth the effort trying to simplify the usages here even if the class ID is not used. The class ID can also be useful when debugging as it is simpler to identify than the type which starts of the same for each object.

Reply:
Maybe there's a misunderstanding, I'm suggesting we remove the ID when we now explicitly supply the Type. The information available during debugging remains the same but it keeps the test complexity equal to before the change. Now, when we specify both, we either have to keep doing it (out of habit or because we think it's the right way to do things) or we will not supply the ID for any new code, making the old code appear to be more complex due to the extra parameter. That's why dropping the redudant parameter is relevant to me.

Reply:
I did revert the places where I think that the class ID was superfluous. But it is difficult the find all places where it is not needed as there are some test that required the class ID to work correctly.

Reply:
Hmm... I don't remember the details from this review, it's likely just the current file, so we could add the classid fallback as type.Name when classid == null and if that doesn't introduce any new errors, just clean up the changed lines in the current file? Otherwise, we know that we can't simply add a fallback and thus have to keep things as they are now.